kernel/Kconfig

```diff
@@ -799,7 +799,6 @@ config SYS_CLOCK_TICKS_PER_SEC
 
 config SYS_CLOCK_HW_CYCLES_PER_SEC
 	int "System clock's h/w timer frequency"
-	depends on !TIMER_READS_ITS_FREQUENCY_AT_RUNTIME
```

Comment:
This is sort of an abuse.  The idea behind the dependency is to prevent apps from accidentally using the "static" clock rate on devices where it might be wrong because the actual rate is selected by the hardware.